api/v1alpha1/networkfence_types.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FencingOperationResult string
+
+const (
+	// FencingOperationResultSucceeded represents the Succeeded operation state.
+	FencingOperationResultSucceeded FencingOperationResult = "Succeeded"
+
+	// FencingOperationResultFailed represents the Failed operation state.
+	FencingOperationResultFailed FencingOperationResult = "Failed"
+)
+
+// SecretSpec defines the secrets to be used for the network fencing operation.
+type SecretSpec struct {
+	// Name specifies the name of the secret.
+	Name string `json:"name,omitempty"`
+
+	// Namespace specifies the namespace in which the secret
+	// is located.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// NetworkFenceSpec defines the desired state of NetworkFence
+type NetworkFenceSpec struct {
+	// Driver contains  the name of CSI driver.
+	// +kubebuilder:validation:Required
+	Driver string `json:"driver"`
+
+	// Cidr contains a CIDR block, which is required to be fenced.
+	// +kubebuilder:validation:Required
+	Cidr string `json:"cidr"`
```

Comment:
do you want this as a single `CIDR`, or a list?

Reply:
In case multiple CIDR blocks are required to be network fenced, one Custom Resource per CIDR block is required so as to keep the operational complexity low.
As each CIDR block then has its own range of IPs; due to which keeping multiple CIDRs in one CR will highly increase nested complexity. 

Although, in the future, adding multiple CIDRs to a CR is definitely a goal.

Reply:
_to keep the operational complexity low_ is not a good reason to make the interface less user-friendly. As the CSI-Addons specification offers a list of CIDRs, it probably is better to follow that approach.

Reply:
Thanks for the suggestion! Going with list of blocks :+1:  Updated now!